### PR TITLE
Upgrade django-field-audit to 1.2.8

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

[Changelog](https://github.com/dimagi/django-field-audit/blob/main/CHANGELOG.md)

The real motivation behind this is to include the migration change made in https://github.com/dimagi/django-field-audit/pull/30 to support Django 4.2 (avoids creating an unnecessary migration).

### QA Plan

No

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/models_role.py [hunk 1]
```diff
@@ -161,7 +161,12 @@ def to_json(self):
     def set_permissions(self, permission_infos):
         def _clear_query_cache():
             try:
-                self.refresh_from_db(fields=["rolepermission_set"])
+                # There is a bug in refresh_from_db when specifying fields that results in this error:
+                # RuntimeError: Set changed size during iteration
+                # Once on a version of Django that includes the change made for
```
Comment (millerdev) [inline corehq/apps/users/models_role.py:166]: Are you keeping this on your radar, or how will we be reminded to do this?

**Review by millerdev — APPROVED** [review]
Kicked test runner that failed with `LoopExit: This operation would block forever`

File: corehq/apps/users/models_role.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/users/models_role.py:166]: https://docs.google.com/spreadsheets/d/1t1LWSlvxzN7WwCFbFBmisM1p33PBIfbqXhm6vJClPeg/edit#gid=663153186

**Comment by gherceg** [comment]
I confirmed there were no other usages of `refresh_from_db` with specific fields for audited models.

**Pull request merged**